frameworks/Crystal/crystal/server.cr

```diff
@@ -30,11 +30,17 @@ server = HTTP::Server.new("0.0.0.0", 8080) do |context|
     response.status_code = 200
     response.headers["Content-Type"] = "application/json"
 
-    results = (1..sanitized_query_count(request)).map do
-      random_world
+    JSON.build(response) do |json|
+      json.array do
+        (1..sanitized_query_count(request)).each do
+          world = random_world          
```

Comment:
Simply call `random_world.to_json(json)` and it'll work, and be *much* cleaner. See [here](https://github.com/crystal-lang/crystal/blob/c6e2077a934c84184844c53d10dee31847e1713b/src/json/to_json.cr#L8).